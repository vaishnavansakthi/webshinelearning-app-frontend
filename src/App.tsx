import { BrowserRouter as Router, Routes, Route } from "react-router-dom"
import { ForgotPassword, ResetPassword, Login, SignUp, SendOtp } from "./pages"
import { Layout } from "./components/templates"
import { useEffect } from "react"
import axios from "axios";

function App() {

  useEffect(() => {
    axios.get(`${import.meta.env.VITE_API_URL}/user/get-all-user`, {
      headers: {
        'x-api-key': `${import.meta.env.VITE_X_API_Key}`,
        'Authorization': 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImQ2ZTYxYTZkLWRmM2UtNGQwOC1iMTViLTc4YjExOTA4OThkMyIsInVzZXJuYW1lIjoidmFpc2huYXZhbiIsImVtYWlsIjoidmFpc2huYXZhbm01QGdtYWlsLmNvbSIsInJvbGUiOiJhZG1pbiIsImlhdCI6MTcwOTczOTMzOH0._XBLzmyVkWXh69xisi1rNmN6LavIDuq-Y4A0xgSU6Xs'
      }
      ,
    }).then((res: any) => {
      console.log(res.data)
    })
  }, [])

  return (
    <Router>
      <Layout>
        <Routes>
          <Route path="/" element={<Login />} />
          <Route path="/signup" element={<SignUp />} />
          <Route path="/forgot-password" element={<ForgotPassword />} />
          <Route path="/reset-password" element={<ResetPassword />} />
          <Route path="/verify-otp" element={<SendOtp />} />
        </Routes>
      </Layout>
    </Router>
  )
}

export default App
