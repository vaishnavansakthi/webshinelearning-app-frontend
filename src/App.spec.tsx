
import TestRenderer from 'react-test-renderer';
import App from './App';

test('Renders the main page', () => {
  const component = TestRenderer.create(<App />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
