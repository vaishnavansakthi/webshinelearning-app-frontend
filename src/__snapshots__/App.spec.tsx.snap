// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the main page 1`] = `
[
  <header
    className="bg-white mb-10 border bottom-3 border-gray-100"
  >
    <nav
      aria-label="Global"
      className="px-10 py-5 flex items-center justify-between w-full"
    >
      <div
        className="flex lg:flex-1"
      >
        <a
          className="-m-1.5 p-1.5"
          href="/"
        >
          <span
            className="text-black text-lg"
          >
            <span
              className="text-2xl font-mono"
            >
              W
            </span>
            ebshine 
            <span
              className="text-2xl font-mono"
            >
              T
            </span>
            alents.
          </span>
        </a>
      </div>
      <div
        className="flex lg:hidden"
      >
        <button
          className="-m-2.5 inline-flex items-center justify-center rounded-md p-2.5 text-gray-700"
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Open main menu
          </span>
          <svg
            aria-hidden="true"
            className="h-6 w-6"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
          >
            <path
              d="M3.75 6.75h16.5M3.75 12h16.5m-16.5 5.25h16.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </button>
      </div>
      <div
        className="hidden lg:flex lg:gap-x-12"
      >
        <div
          className="relative"
        >
          <button
            aria-expanded={false}
            className="flex items-center gap-x-1 text-sm font-semibold leading-6 text-gray-900 "
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            type="button"
          >
            Product
            <svg
              aria-hidden="true"
              className="h-5 w-5 flex-none text-gray-400"
              fill="currentColor"
              viewBox="0 0 20 20"
            >
              <path
                clip-rule="evenodd"
                d="M5.23 7.21a.75.75 0 011.06.02L10 11.168l3.71-3.938a.75.75 0 111.08 1.04l-4.25 4.5a.75.75 0 01-1.08 0l-4.25-4.5a.75.75 0 01.02-1.06z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
        <div
          className="relative"
        >
          <button
            aria-expanded={false}
            className="flex items-center gap-x-1 text-sm font-semibold leading-6 text-gray-900 "
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            type="button"
          >
            Features
            <svg
              aria-hidden="true"
              className="h-5 w-5 flex-none text-gray-400"
              fill="currentColor"
              viewBox="0 0 20 20"
            >
              <path
                clip-rule="evenodd"
                d="M5.23 7.21a.75.75 0 011.06.02L10 11.168l3.71-3.938a.75.75 0 111.08 1.04l-4.25 4.5a.75.75 0 01-1.08 0l-4.25-4.5a.75.75 0 01.02-1.06z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
        <a
          className="text-sm font-semibold leading-6 text-gray-900"
          href="#"
        >
          Marketplace
        </a>
        <a
          className="text-sm font-semibold leading-6 text-gray-900"
          href="#"
        >
          Company
        </a>
      </div>
      <div
        className="hidden lg:flex lg:flex-1 lg:justify-end"
      >
        <a
          className="text-sm font-semibold leading-6 text-gray-900"
          href="/signup"
        >
          Sign Up 
          <span
            aria-hidden="true"
          >
            →
          </span>
        </a>
      </div>
    </nav>
  </header>,
  <section
    className=""
  >
    <div
      className="flex flex-col items-center justify-center"
    >
      <div
        className="w-full bg-white rounded-lg shadow dark:border md:mt-0 sm:p-2 max-w-[340px] sm:max-w-md xl:p-0"
      >
        <div
          className="p-6 space-y-4 md:space-y-6 sm:p-8"
        >
          <button
            className="
        bg-white
         p-4 border-[1px] border-gray-200 shadow-inner rounded-md
         text-sm p-1
         text-gray-400 w-full font-medium text-lg rounded-lg px-6 py-2.5 text-center md:mr-5 mb-3 md:mb-0 inline-flex items-center justify-center tracking-wide undefined
        "
          >
            <svg
              className="w-4 h-4 mr-3"
              id="google"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M113.47 309.408 95.648 375.94l-65.139 1.378C11.042 341.211 0 299.9 0 256c0-42.451 10.324-82.483 28.624-117.732h.014L86.63 148.9l25.404 57.644c-5.317 15.501-8.215 32.141-8.215 49.456.002 18.792 3.406 36.797 9.651 53.408z"
                fill="#fbbb00"
              />
              <path
                d="M507.527 208.176C510.467 223.662 512 239.655 512 256c0 18.328-1.927 36.206-5.598 53.451-12.462 58.683-45.025 109.925-90.134 146.187l-.014-.014-73.044-3.727-10.338-64.535c29.932-17.554 53.324-45.025 65.646-77.911h-136.89V208.176h245.899z"
                fill="#518ef8"
              />
              <path
                d="m416.253 455.624.014.014C372.396 490.901 316.666 512 256 512c-97.491 0-182.252-54.491-225.491-134.681l82.961-67.91c21.619 57.698 77.278 98.771 142.53 98.771 28.047 0 54.323-7.582 76.87-20.818l83.383 68.262z"
                fill="#28b446"
              />
              <path
                d="m419.404 58.936-82.933 67.896C313.136 112.246 285.552 103.82 256 103.82c-66.729 0-123.429 42.957-143.965 102.724l-83.397-68.276h-.014C71.23 56.123 157.06 0 256 0c62.115 0 119.068 22.126 163.404 58.936z"
                fill="#f14336"
              />
            </svg>
            Sign in with Google
          </button>
          <div
            className="flex items-center justify-center my-6"
          >
            <hr
              className="border-t border-gray-300 flex-grow"
            />
            <p
              className="mx-4 text-gray-400"
            >
              or
            </p>
            <hr
              className="border-t border-gray-300 flex-grow"
            />
          </div>
          <form
            action="#"
            className="space-y-4 md:space-y-6"
          >
            <div>
              <label
                className="block mb-2 text-sm font-medium text-gray-900 text-left text-gray-400"
                htmlFor="email"
              >
                Your email
              </label>
              <div
                className="mb-5  flex justify-center"
              >
                <input
                  className=" rounded-lg false bg-gray-50 border w-[400px] text-[16px] border-gray-300 focus:outline-blue-500 rounded-none text-gray-900 text-sm p-2.5 disabled:bg-gray-300 disabled:cursor-not-allowed undefined"
                  id="email"
                  placeholder="name@company.com"
                  style={
                    {
                      "borderRadius": "5px",
                      "display": "inline-block",
                      "verticalAlign": "top",
                    }
                  }
                  type="email"
                />
              </div>
            </div>
            <div>
              <label
                className="block mb-2 text-sm font-medium text-gray-900 text-left text-gray-400"
                htmlFor="password"
              >
                Password
              </label>
              <div
                className="mb-5  flex justify-center"
              >
                <input
                  className=" rounded-lg false bg-gray-50 border w-[400px] text-[16px] border-gray-300 focus:outline-blue-500 rounded-none text-gray-900 text-sm p-2.5 disabled:bg-gray-300 disabled:cursor-not-allowed undefined"
                  id="email"
                  placeholder="••••••••"
                  style={
                    {
                      "borderRadius": "5px",
                      "display": "inline-block",
                      "verticalAlign": "top",
                    }
                  }
                  type="password"
                />
              </div>
            </div>
            <div
              className="flex items-center justify-between"
            >
              <div
                className="flex items-start"
              >
                <div
                  className="flex items-center h-5"
                >
                  <input
                    aria-describedby="remember"
                    className="w-4 h-4 border border-gray-300 rounded bg-gray-50 focus:ring-3 focus:ring-primary-300 dark:bg-gray-700 dark:border-gray-600 dark:focus:ring-primary-600 dark:ring-offset-gray-800"
                    id="remember"
                    type="checkbox"
                  />
                </div>
                <div
                  className="ml-3 text-sm"
                >
                  <label
                    className="block mb-2 text-sm font-medium text-gray-900 text-left text-gray-400"
                    htmlFor="remember"
                  >
                    Remember me
                  </label>
                </div>
              </div>
              <a
                className="text-sm font-medium text-primary-600 hover:underline text-white"
                href="#"
              >
                Forgot password?
              </a>
            </div>
            <button
              className="
        bg-blue-500 hover:bg-blue-600
         
         text-md p-2
         text-white w-full font-medium text-lg rounded-lg px-6 py-2.5 text-center md:mr-5 mb-3 md:mb-0 inline-flex items-center justify-center tracking-wide tracking-wide leading-normal
        "
            >
              Login
            </button>
            <p
              className="text-sm font-light text-gray-500"
            >
              Don’t have an account yet?
               
              <a
                className="text-md font-medium text-primary-600 hover:underline tracking-wide leading-normal"
                href="/signup"
                style={
                  {
                    "color": undefined,
                  }
                }
              >
                Sign Up
              </a>
            </p>
          </form>
        </div>
      </div>
    </div>
  </section>,
  <footer
    className="bg-[#0A0E1F] mt-8"
  >
    <div
      className="mx-auto w-full max-w-screen-2xl p-10 py-6 lg:py-8"
    >
      <div
        className="sm:flex sm:items-center sm:justify-between"
      >
        <span
          className="text-sm text-gray-500 sm:text-center dark:text-gray-400"
        >
          © 2023
           
          <a
            className="hover:underline"
            href="https://flowbite.com/"
          >
            WebshineLearning™
          </a>
          . All Rights Reserved.
        </span>
        <div
          className="flex mt-4 sm:justify-center sm:mt-0"
        >
          <a
            className="text-gray-500 hover:text-gray-900 dark:hover:text-white"
            href="#"
          >
            <svg
              aria-hidden="true"
              className="w-4 h-4"
              fill="currentColor"
              viewBox="0 0 8 19"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M6.135 3H8V0H6.135a4.147 4.147 0 0 0-4.142 4.142V6H0v3h2v9.938h3V9h2.021l.592-3H5V3.591A.6.6 0 0 1 5.592 3h.543Z"
                fill-rule="evenodd"
              />
            </svg>
            <span
              className="sr-only"
            >
              Facebook page
            </span>
          </a>
          <a
            className="text-gray-500 hover:text-gray-900 dark:hover:text-white ms-5"
            href="#"
          >
            <svg
              aria-hidden="true"
              className="w-4 h-4"
              fill="currentColor"
              viewBox="0 0 21 16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16.942 1.556a16.3 16.3 0 0 0-4.126-1.3 12.04 12.04 0 0 0-.529 1.1 15.175 15.175 0 0 0-4.573 0 11.585 11.585 0 0 0-.535-1.1 16.274 16.274 0 0 0-4.129 1.3A17.392 17.392 0 0 0 .182 13.218a15.785 15.785 0 0 0 4.963 2.521c.41-.564.773-1.16 1.084-1.785a10.63 10.63 0 0 1-1.706-.83c.143-.106.283-.217.418-.33a11.664 11.664 0 0 0 10.118 0c.137.113.277.224.418.33-.544.328-1.116.606-1.71.832a12.52 12.52 0 0 0 1.084 1.785 16.46 16.46 0 0 0 5.064-2.595 17.286 17.286 0 0 0-2.973-11.59ZM6.678 10.813a1.941 1.941 0 0 1-1.8-2.045 1.93 1.93 0 0 1 1.8-2.047 1.919 1.919 0 0 1 1.8 2.047 1.93 1.93 0 0 1-1.8 2.045Zm6.644 0a1.94 1.94 0 0 1-1.8-2.045 1.93 1.93 0 0 1 1.8-2.047 1.918 1.918 0 0 1 1.8 2.047 1.93 1.93 0 0 1-1.8 2.045Z"
              />
            </svg>
            <span
              className="sr-only"
            >
              Discord community
            </span>
          </a>
          <a
            className="text-gray-500 hover:text-gray-900 dark:hover:text-white ms-5"
            href="#"
          >
            <svg
              aria-hidden="true"
              className="w-4 h-4"
              fill="currentColor"
              viewBox="0 0 20 17"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M20 1.892a8.178 8.178 0 0 1-2.355.635 4.074 4.074 0 0 0 1.8-2.235 8.344 8.344 0 0 1-2.605.98A4.13 4.13 0 0 0 13.85 0a4.068 4.068 0 0 0-4.1 4.038 4 4 0 0 0 .105.919A11.705 11.705 0 0 1 1.4.734a4.006 4.006 0 0 0 1.268 5.392 4.165 4.165 0 0 1-1.859-.5v.05A4.057 4.057 0 0 0 4.1 9.635a4.19 4.19 0 0 1-1.856.07 4.108 4.108 0 0 0 3.831 2.807A8.36 8.36 0 0 1 0 14.184 11.732 11.732 0 0 0 6.291 16 11.502 11.502 0 0 0 17.964 4.5c0-.177 0-.35-.012-.523A8.143 8.143 0 0 0 20 1.892Z"
                fill-rule="evenodd"
              />
            </svg>
            <span
              className="sr-only"
            >
              Twitter page
            </span>
          </a>
          <a
            className="text-gray-500 hover:text-gray-900 dark:hover:text-white ms-5"
            href="#"
          >
            <svg
              aria-hidden="true"
              className="w-4 h-4"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M10 .333A9.911 9.911 0 0 0 6.866 19.65c.5.092.678-.215.678-.477 0-.237-.01-1.017-.014-1.845-2.757.6-3.338-1.169-3.338-1.169a2.627 2.627 0 0 0-1.1-1.451c-.9-.615.07-.6.07-.6a2.084 2.084 0 0 1 1.518 1.021 2.11 2.11 0 0 0 2.884.823c.044-.503.268-.973.63-1.325-2.2-.25-4.516-1.1-4.516-4.9A3.832 3.832 0 0 1 4.7 7.068a3.56 3.56 0 0 1 .095-2.623s.832-.266 2.726 1.016a9.409 9.409 0 0 1 4.962 0c1.89-1.282 2.717-1.016 2.717-1.016.366.83.402 1.768.1 2.623a3.827 3.827 0 0 1 1.02 2.659c0 3.807-2.319 4.644-4.525 4.889a2.366 2.366 0 0 1 .673 1.834c0 1.326-.012 2.394-.012 2.72 0 .263.18.572.681.475A9.911 9.911 0 0 0 10 .333Z"
                fill-rule="evenodd"
              />
            </svg>
            <span
              className="sr-only"
            >
              GitHub account
            </span>
          </a>
          <a
            className="text-gray-500 hover:text-gray-900 dark:hover:text-white ms-5"
            href="#"
          >
            <svg
              aria-hidden="true"
              className="w-4 h-4"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M10 0a10 10 0 1 0 10 10A10.009 10.009 0 0 0 10 0Zm6.613 4.614a8.523 8.523 0 0 1 1.93 5.32 20.094 20.094 0 0 0-5.949-.274c-.059-.149-.122-.292-.184-.441a23.879 23.879 0 0 0-.566-1.239 11.41 11.41 0 0 0 4.769-3.366ZM8 1.707a8.821 8.821 0 0 1 2-.238 8.5 8.5 0 0 1 5.664 2.152 9.608 9.608 0 0 1-4.476 3.087A45.758 45.758 0 0 0 8 1.707ZM1.642 8.262a8.57 8.57 0 0 1 4.73-5.981A53.998 53.998 0 0 1 9.54 7.222a32.078 32.078 0 0 1-7.9 1.04h.002Zm2.01 7.46a8.51 8.51 0 0 1-2.2-5.707v-.262a31.64 31.64 0 0 0 8.777-1.219c.243.477.477.964.692 1.449-.114.032-.227.067-.336.1a13.569 13.569 0 0 0-6.942 5.636l.009.003ZM10 18.556a8.508 8.508 0 0 1-5.243-1.8 11.717 11.717 0 0 1 6.7-5.332.509.509 0 0 1 .055-.02 35.65 35.65 0 0 1 1.819 6.476 8.476 8.476 0 0 1-3.331.676Zm4.772-1.462A37.232 37.232 0 0 0 13.113 11a12.513 12.513 0 0 1 5.321.364 8.56 8.56 0 0 1-3.66 5.73h-.002Z"
                fill-rule="evenodd"
              />
            </svg>
            <span
              className="sr-only"
            >
              Dribbble account
            </span>
          </a>
        </div>
      </div>
    </div>
  </footer>,
]
`;
